bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/gnirs/InstGNIRS.java

```diff
@@ -364,6 +371,41 @@ public void setOverrideAcqObsWavelength(boolean newValue) {
 
     // ------------------------------------------------------------------------
 
+    /**
+     * Get the Hartmann mask value.
+     * NOTE: these are not available in the public interface but must be present for properties and to be iterable.
+     */
+    public HartmannMask getHartmannMask() {
+        return _hartmannMask;
+    }
+
+    /**
+     * Set the Hartmann mask.
+     */
+    public void setHartmannMask(HartmannMask newValue) {
+        HartmannMask oldValue = getHartmannMask();
+        if (oldValue != newValue) {
+            _hartmannMask = newValue;
+            firePropertyChange(HARTMANN_MASK_PROP.getName(), oldValue, newValue);
+        }
+    }
+
+    // ------------------------------------------------------------------------
+
+
+    /**
+     * Return the (read-only, static) value for focus.
+     */
+    public Focus getFocus() {
+        return new Focus();
+    }
+
+    public void setFocus(Focus focus) {
+        // Required for reflection
+    }
```

Comment:
I'm not sure about this part.  When you say "read-only, static" you just mean that it is included in the instrument component but isn't edited there?  It does appear, and is editable, in the GNIRS iterator though?

I worry because there is machinery around syncing the first step of the iterator with the static component.  With this implementation it will not be in sync which may cause problems.  I realize there is an outstanding task to remove that syncing but for now perhaps we should just add a `Focus` property and get/set it like everything else.

Reply:
I admit that I copied this verbatim from the NIRI focus editor. The only thing that is a little strange is that it is not possible to edit the first value (it immediately reverts to 'best focus'), perhaps because of the synchronization that you described, which is a minor annoyance but not a hindrance to using this for engineering.

Reply:
Thanks! That works perfectly!